lib/puppet/provider/service/init.rb

```diff
@@ -152,6 +152,15 @@ def restartcmd
     (@resource[:hasrestart] == :true) && [initscript, :restart]
   end
 
+  def texecute(type, command, fof = true, squelch = false, combine = true)
+    if type =~ /^start/
+      if Facter.value(:operatingsystem) == "Solaris"
+        command =  ["/usr/bin/ctrun -l none", command].flatten.join(" ")
```

Comment:
Now that I say that, I'm wondering why we don't have new service provider, whose parent is `:init` and is confined to operatingsystem `Solaris` and then the override logic can live there?

Also do we want to be using osfamily instead of operatingsystem?